src/main/webapp/scripts/map.js

```diff
@@ -80,12 +88,68 @@ window.initMap = function () {
         strokeOpacity: 0.8,
         strokeWeight: 2
       });
+      // Once cards are retrieved...
+      // Add vendor's markers
+      addVendorsToMap(map);
     },
-    (error) => {
-      if (error.code == error.PERMISSION_DENIED)
-        alert('Permission denied to access location');
-    });
+      (error) => {
+        if (error.code == error.PERMISSION_DENIED)
+          alert('Permission denied to access location');
+      });
   } else {
     alert('This browser does not support location');
   }
+};
+
+/**
+ * Function that adds vendor marks to the map
+ */
+const addVendorsToMap = (map) => {
+  const salecardTemplate = document.getElementById('salecard-template');
+  const salecardsContainer = document.getElementById('salecards-container');
+
+  //Clean previously retrieved cards
+  salecardsContainer.textContent = '';
```

Comment:
Hey, Abraham, I think `innerHTML` was okay here, since you were just cleaning the container. 
I guess you tested that `textContent` was indeed cleaning the container, so if it works you can leave it like this. :)

Reply:
Yes, it works because it is only text